### DISCUSSION TITLE
query whitout JSON

### DISCUSSION THREAD

**Discussion by PedroColletti** [adonisjs/core#3695] [Help] [answered]
Guys, how do I pass string instead of JSON in a request?

![image](https://user-images.githubusercontent.com/71047937/165825771-7fb36b7a-e3fa-4ac6-81f2-ef9fd0c3454f.png)

I would like to pass a string through the request that would serve as a filter for the search, my code is like this:

![image](https://user-images.githubusercontent.com/71047937/165827065-161025fe-9624-41e7-8c64-94a595cd8114.png)

**Comment by thetutlage** [comment, marked as answer]
Is it valid JSON?

**Reply by PedroColletti** [reply]
Valid JSON would be an object with a key and value, correct?

Example:
{
 "name": "Thomas"
}

But I would like to make a string search request, and the response returns everything related to the search, as in the first image. That code is JAVA. Could you tell me if it is possible to do this in AdonisJS?

**Reply by NerdyLuffy** [reply]
> t code is JAVA. Could you tell me if it is possible to do this in Ad

Yes @PedroColletti , you can pass the `string` as query parameter in the API and check at the controller level, some additional information over here https://docs.adonisjs.com/guides/request#query-string-and-route-params

So change the API to something like `/api/material/search/16?q=2.7X30MM` and now your `console.log(data)` should show something like `{q: '2.7X30MM'}` and you can access that value using `data.q` at the query level hope that helps 👍

**Reply by PedroColletti** [reply]
Thanks a lot for the help. I ended up turning it into get, and going through the URL. Thanks again for the support, guys.
